AWFUL-HACKS.md

```diff
@@ -83,3 +83,30 @@ Can be removed when one of:
 * Either of the above but on unstable, and my frustration with
   threading bugs (but fearless concurrency, David!) outweighs
   my desire to not use nightly.
+
+### Using Eigenclasses for Example Printing
```

Comment:
I've been researching a possible alternative here. At the end of the run the formatter calls dump_failures, which formats the failed examples through the **failure notifications**. The notification uses an exception presenter, which has an option to change the details or extra details formatter. That might provide an in to cleanly inject the Hypothesis values.

I think it's generally made through a private builder, though, so there still would be _some_ monkeypatching? But it would be more contained. May not be worth the effort though.

dump failures https://github.com/rspec/rspec-core/blob/46539bce2b107ac0291916dfb727b8c2fc6f92c0/lib/rspec/core/formatters/protocol.rb#L137
fully_formatted_failed_examples call https://github.com/rspec/rspec-core/blob/41caa12a7aaffdc2395cda142c549ed241542608/lib/rspec/core/formatters/base_text_formatter.rb#L32
definition http://www.rubydoc.info/github/rspec/rspec-core/RSpec/Core/Notifications/ExamplesNotification#fully_formatted_failed_examples-instance_method
FailedExampleNotification https://github.com/rspec/rspec-core/blob/f27f6a107af7e9011c7801cb124eacc06e47b67a/lib/rspec/core/notifications.rb#L211
ExceptionPresenter https://github.com/rspec/rspec-core/blob/master/lib/rspec/core/formatters/exception_presenter.rb

Reply:
Thanks, this is good research! I'll add some of it to the document.

I've actually gone overnight from thinking this solution is awful to thinking that it's surprisingly nice.

Bad:

* It's a weird hack

Good:

* It actually almost *isn't* monkey-patching. To the degree that it is, it's a very small patch to a very localised monkey. No objects aside from this one exception that we are throwing get patched, and it only touches core methods in a way that is entirely API compatible. So it's *very* robust across versions.
* It's mostly agnostic across minitest and rspec and anything else (though FUN FACT I thought I'd check having said that and it turns out that rspec and minitest disagree as to whether they call to_s or inspect)


Now I still think the correct thing to do is to grow some proper APIs in the test runners for doing that, and to use those when they are available, but I think I've almost reconciled myself to this as a long term solution.